docs/core/sdk/file-based-apps.md

```diff
@@ -0,0 +1,406 @@
+---
+title: File-based apps
+description: Learn how to create, build, and run C# applications from a single file without a project file.
+ms.date: 12/05/2025
+ai-usage: ai-assisted
+---
+# File-based apps
+
+**This article applies to:** ✔️ .NET 10 SDK and later versions
+
+File-based apps let you build, run, and publish .NET applications from a single C# file without creating a traditional project file. They offer a lightweight alternative to traditional .NET projects. This approach simplifies development for scripts, utilities, and small applications. The .NET SDK automatically generates the necessary project configuration based on the directives in your source file.
+
+Key benefits include:
+
+- Reduced boilerplate for simple applications.
+- Self-contained source files with embedded configuration.
+- Native AOT publishing enabled by default.
+- Automatic packaging as .NET tools.
+
+In this article, learn how to create, configure, and work with file-based apps effectively.
+
+## Supported directives
+
+File-based apps use directives prefixed with `#:` to configure the build and run the application. Supported directives include: `#:package`, `#:project`, `#:property`, and `#:sdk`. Place these directives at the top of the C# file.
+
+### `#:package`
+
+Adds a NuGet package reference to your application.
+
+```csharp
+#:package Newtonsoft.Json
+#:package Serilog version="3.1.1"
+```
+
+### `#:project`
+
+References another project file or directory that contains a project file.
+
+```csharp
+#:project ../SharedLibrary/SharedLibrary.csproj
+```
+
+### `#:property`
+
+Sets an MSBuild property value.
+
+```csharp
+#:property TargetFramework=net10.0
+#:property PublishAot=false
+```
+
+### `#:sdk`
+
+Specifies the SDK to use. Defaults to `Microsoft.NET.Sdk`.
+
+```csharp
+#:sdk Microsoft.NET.Sdk.Web
+```
+
+## CLI commands
+
+The .NET CLI provides full support for file-based apps through familiar commands.
+
+### Run applications
+
+Run a file-based app directly by using the `dotnet run` command:
+
+```dotnetcli
+dotnet run file.cs
+```
+
+Or use the shorthand syntax:
+
+```dotnetcli
+dotnet file.cs
+```
+
+#### Pass arguments
+
+Pass arguments to your application in several ways:
+
+```dotnetcli
+dotnet run file.cs -- arg1 arg2
+```
+
+Arguments after `--` are passed to your application. Without `--`, arguments go to the `dotnet run` command:
+
+```dotnetcli
+dotnet run file.cs arg1 arg2
+```
+
+However, with the shorthand syntax, all arguments go to your application:
```

Comment:
This is a bit more nuanced, as the `dotnet` CLI itself uses `System.CommandLine` the arguments will be processed by `dotnet` and in some cases this can interfere with your application, e.g. when using response files. For this reason, it's always safest to pass arguments to your application by placing them after the `--`, even when not using the `run` sub-command.